marketplace-service/src/test/java/com/axonivy/market/service/impl/ProductMarketplaceDataServiceImplTest.java

```diff
@@ -0,0 +1,166 @@
+package com.axonivy.market.service.impl;
+
+import com.axonivy.market.BaseSetup;
+import com.axonivy.market.constants.ProductJsonConstants;
+import com.axonivy.market.entity.Product;
+import com.axonivy.market.entity.ProductCustomSort;
+import com.axonivy.market.entity.ProductMarketplaceData;
+import com.axonivy.market.enums.ErrorCode;
+import com.axonivy.market.enums.SortOption;
+import com.axonivy.market.exceptions.model.InvalidParamException;
+import com.axonivy.market.exceptions.model.NotFoundException;
+import com.axonivy.market.model.ProductCustomSortRequest;
+import com.axonivy.market.repository.ProductCustomSortRepository;
+import com.axonivy.market.repository.ProductMarketplaceDataRepository;
+import com.axonivy.market.repository.ProductRepository;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
```

Comment:
we just use 3 api from Assertions & 4 from ArgumentMatchers, avoid import too muck unused thing.